src/guidellm/utils/text.py

```diff
@@ -1,45 +1,108 @@
+
+Provides comprehensive text processing capabilities including cleaning, filtering,
+splitting, loading from various sources, and formatting utilities. Supports loading
+text from URLs, compressed files, package resources, and local files with automatic
+encoding detection. Includes specialized formatting for display values and text
+wrapping operations for consistent presentation across the system.
+"""
+
+from __future__ import annotations
+
 import gzip
 import re
 import textwrap
 from importlib.resources import as_file, files  # type: ignore[attr-defined]
 from pathlib import Path
-from typing import Any, Optional, Union
+from typing import Any
 
 import ftfy
 import httpx
 from loguru import logger
 
 from guidellm import data as package_data
 from guidellm.config import settings
+from guidellm.utils.console import Colors
 
 __all__ = [
+    "MAX_PATH_LENGTH",
     "EndlessTextCreator",
     "clean_text",
     "filter_text",
+    "format_value_display",
     "is_puncutation",
     "load_text",
     "split_text",
     "split_text_list_by_length",
 ]
 
-MAX_PATH_LENGTH = 4096
+MAX_PATH_LENGTH: int = 4096
+
+
+def format_value_display(
+    value: float,
+    label: str,
+    units: str = "",
+    total_characters: int | None = None,
+    digits_places: int | None = None,
+    decimal_places: int | None = None,
+) -> str:
+    """
+    Format a numeric value with units and label for consistent display output.
+
+    Creates standardized display strings for metrics and measurements with
+    configurable precision, width, and color formatting. Supports both
+    fixed-width and variable-width output for tabular displays.
+
+    :param value: Numeric value to format and display
+    :param label: Descriptive label for the value
+    :param units: Units string to append after the value
+    :param total_characters: Total width for right-aligned output formatting
+    :param digits_places: Total number of digits for numeric formatting
+    :param decimal_places: Number of decimal places for numeric precision
+    :return: Formatted string with value, units, and colored label
+    """
+    if decimal_places is None and digits_places is None:
+        formatted_number = f"{value}:.0f"
+    elif digits_places is None:
+        formatted_number = f"{value:.{decimal_places}f}"
+    elif decimal_places is None:
+        formatted_number = f"{value:>{digits_places}f}"
+    else:
+        formatted_number = f"{value:>{digits_places}.{decimal_places}f}"
+
+    result = f"{formatted_number}{units} [{Colors.info}]{label}[/{Colors.info}]"
+
+    if total_characters is not None:
+        total_characters += len(Colors.info) * 2 + 5
+
+        if len(result) < total_characters:
+            result = result.rjust(total_characters)
+
+    return result
 
 
 def split_text_list_by_length(
     text_list: list[Any],
-    max_characters: Union[int, list[int]],
+    max_characters: int | list[int],
```

Comment:
If you do this then we will need to bump the minimum Python version to 3.10. Are we ready for that commitment?

Also applies to several other instances in this file.

I'd say keeping 3.9 would be easier since it's the default RHEL Python version.

Reply:
Unfortunately we can't give up 3.9 until October with EOL. The from future import __annotations__ allows us to use the | where Unions/Optionals would be as well as some other nice features such as all types automatically being string types preventing some more issues

Reply:
Thank you for clarifying. Good that it's possible to opt into that feature for 3.9.